### PR TITLE
Don't memset Constant's buffer if it's about to be filled with data

### PR DESCRIPTION
**Pull request by mateusztabaka**

### CODE REVIEW

File: src/core/src/op/constant.cpp [hunk 1]
```diff
@@ -537,7 +541,7 @@ bool ov::op::v0::Constant::visit_attributes(AttributeVisitor& visitor) {
     bool need_to_reallocate = (m_shape != prev_shape || prev_type != m_element_type);
     if (m_alloc_buffer_on_visit_attributes && need_to_reallocate) {
         // Filling in a fresh constant
-        allocate_buffer();
+        allocate_buffer(true);
```
[the remark this answers is not in the document]
Comment (mateusztabaka) [reply src/core/src/op/constant.cpp:544]: I think you're probably right. I missed that.

[the remark this answers is not in the document]
Comment (ilyachur) [reply src/core/src/op/constant.cpp:544]: Yes right, visit attributes should fill the data.

**Review by ilyachur — COMMENTED** [review]
Do you have any numbers which show improvement from this change?

**Comment by mateusztabaka** [comment]
I have some microbenchmark results for creating a float Constant:


num elements | before (ns) | after (ns) | speedup
-- | -- | -- | --
1024 | 900 | 861 | 1.045296
4096 | 2380 | 2239 | 1.062975
16384 | 9070 | 7350 | 1.234014
65536 | 47114 | 37245 | 1.264975
262144 | 193041 | 152586 | 1.265129
1048576 | 780380 | 610100 | 1.279102
4194304 | 3205258 | 2510327 | 1.276829
16777216 | 51729915 | 44226177 | 1.169667

File: src/core/src/op/constant.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mateusztabaka) [reply src/core/src/op/constant.cpp]: done

**Review by ilyachur — APPROVED** [review]

**Pull request merged**